appsensor-core/src/main/java/org/owasp/appsensor/core/IAppsensorEntity.java

```diff
@@ -0,0 +1,14 @@
+package org.owasp.appsensor.core;
+
+import java.io.Serializable;
+
+/**
+ * Interface denoting methods required to be provided by appsensor entity objects
+ * @author Maik Jäkel(m.jaekel@xsite.de) http://www.xsite.de
+ */
+public interface IAppsensorEntity extends Serializable{
```

Comment:
@jtmelton The interface is necessary to provide a generic way of setting and getting the id from all entity types. This way I could write a generic way of storing and retrieving objects from elastic search. The code using this interface is [AbstractElasticRepostory.java:218](https://github.com/MaikJaek/appsensor/blob/42022784dd04cf205e28a908075973eb53bfaf99/storage-providers/appsensor-storage-nosql-elasticsearch/src/main/java/org/owasp/appsensor/storage/elasticsearch/dao/AbstractElasticRepository.java#L218). 

One would expect that this interface is not necessary, since the objects are automatically serialized / deserialized using a JSON mapper. However, the ES id attribute is not called 'id' but '_id' and therefore not automatic mapping of values is taking place. This, by the way, is also true for MongoDB. The current implementation of MongoDB does not map the auto generated ids to the objects because the corresponding database field is called '_id'.